### ISSUE TITLE
Could not find a part of the path

### ISSUE THREAD

**Issue by rcpassos** [jfversluis/FilePicker-Plugin-for-Xamarin-and-Windows#83]
## Expected Behavior
I'm trying to Pick a file from iOS.

## Actual Behavior
I'm getting this error:
"Could not find a part of the path "/private/var/containers/Bundle/Application/..."

## Specifications

  - NuGet package version: 2.0.81-beta
  - Platform(s): iOS

**Comment by jfversluis** [comment]
Doe the path contain spaces?

**Comment by rcpassos** [comment]
This is the full Exception message:
`Could not find a part of the path "/private/var/containers/Bundle/Application/6A86884A-D30D-4447-A0A2-EB7C5B368037/AppName.iOS.app/file:/private/var/mobile/Containers/Data/Application/A278D189-7189-4DB2-AA28-1F2685407567/tmp/com.bundleid-Inbox/file_name.txt".`

**Comment by rcpassos** [comment]
> Doe the path contain spaces?

No, as you can see in the full exception above

**Comment by rcpassos** [comment]
@jfversluis Do you want more info?

**Comment by jfversluis** [comment]
If you have anything more that would be great

**Comment by softsan** [comment]
we are using v2.0.81-beta
Also getting following error on iOS (file.DataArray):
`System.IO.DirectoryNotFoundException: Could not find a part of the path "/private/var/containers/Bundle/Application/E6F13309-B18A-4905-96CB-55BFD8145F9F/myproject.app/file:/private/var/mobile/Containers/Data/Application/6EB34D7B-B279-4664-A758-FCC4478D095F/tmp/com.myproject-Inbox/sample.pdf".`

**Comment by tehcodez** [comment]
Experiencing this too. Previous version (v2.0.64-beta) had the issue where if the file name contained spaces it would throw  as described in #62. New version (v2.0.81-beta) has this error and seems to throw when selecting any file from cloud storage (i have tried with iCloud and Google Drive).

**Comment by rcpassos** [comment]
Please look at this accepted answer https://stackoverflow.com/questions/46946482/xamarin-ios-file-picker-unable-to-get-selected-file-from-icloud-drive

**Comment by JunielKatarn** [comment]
I also face this issue. Seems like the iOS implementation prepends the path to the app bundle to the file to be imported path.

I forked the repo (skipping the NuGet for now) and applied @rcpassos here:
https://github.com/JunielKatarn/FilePicker-Plugin-for-Xamarin-and-Windows/commit/08347fe0ff0e8caf750883539285e50265433aa4

**Comment by santoshg31** [comment]
I am facing the same problem. Moved over to this latest version of nuget as the previous version 2.0.64-beta had three problems in ios.
1. Word document files were disabled.
2. Error while picking files with spaces 
3. There was no browse button appearing on ipad.

@JunielKatarn   did you just add the dlls to your  core and UI project after making changes and did the above three issues no longer appear as well?

@jfversluis  will there be new nuget version for the fix that @JunielKatarn  had added for this issue ?

**Comment by xinwanc** [comment]
Hi, actually this not only occurs on cloud drive but also occurs on google drive, when i try to get the file path from the file that I selected from Google drive it also shows this error

**Comment by JunielKatarn** [comment]
> did you just add the dlls to your core and UI project after making changes and did the above three issues no longer appear as well

@santoshg31  No. I fully got rid of the pre-built DLLs and added this project as a sub module of my project.
With some minor tweaking, I got it working.
It is lightweight enough so I’m doesnt add a lot of overhead to the build.

**Comment by jcus0006** [comment]
@JunielKatarn thanks for taking the time to comment here with your solution / workaround. This is the closest to a potential solution I have found for my issue/s. Strangely for me; with the latest Beta version 2.0.81 I am getting the same error as the OP of this thread: "Could not find a part of the path "/private/var/containers/Bundle/Application/...", whereas if I downgrade to 2.0.64, I manage to get it to work with an iPhone 6s Plus whereas the iCloud browse is never opened for an iPad. Being a bit pressed against time and not knowing whether there'll be a NuGet update fixing this issue I am attempting to implement your solution in my project.

Can you let me know whether I understood the process of including your project correctly? 

1. I uninstalled the NuGet project for iOS (left the Android as it is working for me). 
2. I downloaded the project from your forked repo and I included the Plugin.FilePicker and Plugin.FilePicker.Abstractions projects into my Xamarin project; 
3. I built the projects (and they built successfully) and then referenced the DLLs from the bin into the Xamarin.iOS project's references.

If I understand correctly including the projects into my Xamarin solution is only required if I am to add any further updates to the source code cause otherwise simply referencing the DLLs would have been sufficient?

Is this the correct way to use your fix? Did I miss any important step?

----------------------------------------------------------------------------
Update:

I could test it out and I got this error: 

> {System.NotImplementedException: This functionality is not implemented in the portable version of this assembly. 
>  You should reference the NuGet package from your main application project in order to reference the platform-specific implementation.

This is the same error I get when I try to use the stable version of the NuGet plugin. Does it mean that I need to create a dependency service for both platforms?

Or are these using lines not referring to the locally referenced DLLs as is?:

```
using Plugin.FilePicker;
using Plugin.FilePicker.Abstractions;
```

**Comment by vividos** [comment]
This is fixed in NuGet package 2.0.106-beta. Please try out. The FileData class now returns the path without %20 and file:// prefix, so opening with FileStream should just work. Even better, you can just get a stream by using GetStream(). Please try it out with the mentioned package. Thanks!
